subsys/bluetooth/host/gatt_gap_svc_validate.c

```diff
@@ -0,0 +1,55 @@
+/* gatt_gap_svc_validate.c - GAP service inside GATT validation functions */
+
+/*
+ * Copyright (c) 2025 Koppel Electronic
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/logging/log.h>
+
+
+#define MODULE gatt_gap_validate
+
+LOG_MODULE_REGISTER(MODULE, CONFIG_GATT_GAP_SVC_VALIDATE_LOG_LEVEL);
+
+
+static uint8_t gap_service_check_cb(const struct bt_gatt_attr *attr,
+				    uint16_t handle,
+				    void *user_data)
+{
+	size_t *cnt = user_data;
```

Comment:
Consider if `cnt` should be changed to `bool found`; do we care about how many services we find, or do we simply care that we found one and only one?